apps/dotcom/client/src/tla/pages/local.tsx

```diff
@@ -11,19 +11,22 @@ import { getFilePath } from '../utils/urls'
 export function Component() {
 	const app = useMaybeApp()
 	const navigate = useNavigate()
+	const creatingFile = useRef(false)
 
 	useEffect(() => {
 		if (!app) return
 		if (app.getUserRecentFiles().length === 0) {
 			app.createFile().then((res) => {
 				if (res.ok) {
+					creatingFile.current = true
 					navigate(getFilePath(res.value.file.id), { state: { mode: 'create' } })
```

Comment:
This navigate does not kick in immediately, it ran the logic for redirecting to the recent file, which then removed the `creating` mode from the navigation.